generators/entity/templates/server/src/main/java/package/domain/_Entity.java

```diff
@@ -58,8 +58,9 @@
 <% } -%>
 <% if (databaseType == 'sql') { -%>
 @Entity
-@Table(name = "<%= entityTableName %>")<% if (hibernateCache != 'no') { %>
-@Cache(usage = CacheConcurrencyStrategy.NONSTRICT_READ_WRITE)<% } %><% } %><% if (databaseType == 'mongodb') { %>
+@Table(name = "<%= entityTableName %>")
+<% if (hibernateCache != 'no') { if (hibernateCache == 'infinispan') { %>@Cache(usage = CacheConcurrencyStrategy.READ_WRITE) <% } else { %>
```

Comment:
since this part is becoming complex may be we can switch to using `<%_` to have newline slurped
so
```
<%_ if (hibernateCache != 'no') { if (hibernateCache == 'infinispan') { _%>
@Cache(usage = CacheConcurrencyStrategy.READ_WRITE)
<%_ } else { _%>
@Cache(usage = CacheConcurrencyStrategy.NONSTRICT_READ_WRITE)
<%_ } } 
} if (databaseType == 'mongodb') { _%>
```